render/dom-for.js

```diff
@@ -0,0 +1,25 @@
+"use strict"
+
+var delayedRemoval = new WeakMap
+module.exports.delayedRemoval = delayedRemoval
+
+module.exports.domFor = function *domFor(vnode, {generation} = {generation: undefined}) {
+	let {dom, domSize} = vnode
+	if (dom != null) {
+		if (domSize == null) {
+			if (delayedRemoval.get(dom) === generation) {
+				yield dom
+			}
+		} else {
+			let i = 0, next
+			while (i < domSize) {
+				next = dom.nextSibling
+				if (delayedRemoval.get(dom) === generation) {
+					yield dom
+					i++
+				}
+				dom = next
+			}
+		}
+	}
+}
```

Comment:
I struggle a bit with the complexity of. What do you think to golfing it down to this? Lints well too.

```suggestion
function *domFor({dom, domSize}, {generation} = {generation: undefined}) {
	if(dom == null)
		return

	if (domSize == null) {
		if (delayedRemoval.get(dom) === generation)
			yield dom
	}
	else while (domSize) {
		if (delayedRemoval.get(dom) === generation) {
			yield dom
			domSize--
		}
		dom = dom.nextSibling
	}
}
```

Reply:
There's a general pattern of camelCasing in module filenames: the module and the test should both call this the same as the function itself — `domFor` — IMO

Reply:
Re. CamelCasing, I misremembered and didn't verify...

Re. `domFor()` details the `next = dom.nextSibling` bit is necessary, since `domFor` is used internally to remove nodes.

I went for the `if (dom != null)` route because it is slightly shorter, but minifiers probably do that kind of transform under the hood. An early return would be fine by me, and `domSize--` is good too.

Reply:
Oh the recursion scenario hadn't occurred to me. Ought that to be tested for? The code I posted passes all tests.

Reply:
Err, it shouldn't... `domFor` is used to remove fragments, and if the yielded node is removed, its `.nextSibling` becomes null...

I get five bail outs here, with "_Failed to execute removeChild, parameter is not of type 'Node'_".

Reply:
[Failures confirmed!](https://github.com/MithrilJS/mithril.js/runs/6843951633?check_suite_focus=true#step:5:11)

Reply:
[Fixed my branch](https://github.com/MithrilJS/mithril.js/actions/runs/2480387255) - [the PR on your repo](https://github.com/pygy/mithril.js/pull/3) is good to go now.

Reply:
This was merged, resolving